ural/ensure_protocol.py

```diff
@@ -0,0 +1,17 @@
+# =============================================================================
+# Ural URL Ensure Protocol Function
+# =============================================================================
+#
+# A function checking if the url features a protocol, and adding one if there is none.
+#
+import re
+
+PROTOCOL_RE = re.compile('^[^:]*:?//')
```

Comment:
same thing, and maybe this regexp could be factorised somewhere?

Reply:
@Yomguithereal des idées sur où factoriser cette regexp ? Un fichier dédié ?

Reply:
Oui, un fichier dédié genre `regexes.py` ou `patterns.py`. A terme on pourrait même l'exposer et le documenter si ça peut être utile.